examples/server-communication__offline/cypress/integration/offline-spec.js

```diff
@@ -1,177 +1,87 @@
-  beforeEach(goOnline)
-  afterEach(goOnline)
-
-  it('shows network status', () => {
-    cy.visit('/')
-    cy.contains('#network-status', 'online')
-    .wait(1000) // for demo purpose
-
-    goOffline()
-    cy.contains('#network-status', 'offline')
-    .wait(1000) // for demo purpose
-  })
-
-  it('shows error if we stub the network call', () => {
-    assertOnline()
-    cy.visit('/')
-    cy.intercept(`${url}*`, { forceNetworkError: true }).as('users')
-    cy.get('#load-users').click()
-    cy.contains('#users', 'Problem fetching users Failed to fetch')
-
-    // cannot wait for the intercept that forces network error
-    // https://github.com/cypress-io/cypress/issues/9062
-    // cy.wait('@users', { timeout: 1000 }) // the network call happens
-  })
-
-  it('shows error trying to fetch users in offline mode', () => {
-    cy.visit('/')
-    assertOnline()
-
-    // since this call returns a promise, must tell Cypress to wait
-    // for it to be resolved
-    goOffline()
-    assertOffline()
+  context('when online', () => {
+    beforeEach(() => {
+      cy.visit('/', {
+        onBeforeLoad(win) {
+          // use window.navigator.onLine property to
+          // mock browser is offline or online
+          // https://caniuse.com/online-status
+          cy.stub(win.navigator, 'onLine', true).as('online')
+        }
+      })
+      cy.intercept('*').as('users')
 
-    cy.get('#load-users').click()
-    cy.contains('#users', 'Problem fetching users Failed to fetch')
+      cy.wrap(window).its('navigator.onLine').should('be.true')
+    })
 
-    // now let's go back online and fetch the users
-    goOnline()
-    assertOnline()
-    cy.get('#load-users').click()
-    cy.get('.user').should('have.length', 3)
+    it('shows network status as online', () => {
+      cy.contains('#network-status', 'online')
+    })
+  
+    it('fetches users', () => {
+      // let's spy on the "fetch" method the app calls
+      cy.window().then((w) => cy.spy(w, 'fetch').withArgs(`${url}?_limit=3`).as('fetchUsers'))
+      cy.get('#load-users').click()
+      cy.get('.user').should('have.length', 3)
+      cy.get('@fetchUsers').should('have.been.calledOnce')
+    })
   })
 
-  it('makes fetch request when offline', () => {
-    cy.visit('/')
-
-    goOffline()
-    assertOffline()
-
-    // let's spy on the "fetch" method the app calls
-    cy.window().then((w) => cy.spy(w, 'fetch').withArgs(`${url}?_limit=3`).as('fetchUsers'))
-
-    cy.get('#load-users').click()
-    cy.get('@fetchUsers').should('have.been.calledOnce')
-
-    // now let's go back online and fetch the users
-    goOnline()
-    assertOnline()
-    cy.get('#load-users').click()
-    cy.get('.user').should('have.length', 3)
-    cy.get('@fetchUsers').should('have.been.calledTwice')
-  })
+  context('when offline', () => {
+    beforeEach(() => {
+      cy.visit('/', {
+        onBeforeLoad(win) {
+          cy.stub(win.navigator, 'onLine', false).as('online')
+        }
```

Comment:
man, `navigator.onLine` is so silly, as are `ononline` and `onoffline`... in 99% of cases, they are useless, since `onLine` is always true as long as there's any type of network adapter... even a LAN adapter with no internet access

always gets me.

from MDN:

> In Chrome and Safari, if the browser is not able to connect to a local area network (LAN) or a router, it is offline; all other conditions return true. So while you can assume that the browser is offline when it returns a false value, you cannot assume that a true value necessarily means that the browser can access the internet. You could be getting false positives, such as in cases where the computer is running a virtualization software that has virtual ethernet adapters that are always "connected." Therefore, if you really want to determine the online status of the browser, you should develop additional means for checking

https://developer.mozilla.org/en-US/docs/Web/API/NavigatorOnLine/onLine

/rant